documentation/c10-classdesign.sil

```diff
@@ -261,8 +261,6 @@ The latter acts as a setter or getter, so a minimal code will usually look as fo
 
 \begin[type=autodoc:codeblock]{raw}
 function class:declareOptions ()
-  base.declareOptions(self) -- Note: support parent class options
```

Comment:
base --> plain -> A --> B
B no longer has to call the parent method?

Reply:
(Also I don't see how it related to "language support")

Reply:
> base --> plain -> A --> B B no longer has to call the parent method?

Correct. One should never use base (prototype) modules directly, but always something that inherits from them. Anything inheriting *directly* from a base class doesn't have to worry about calling the parent method(s) other than `base._init(self)`, all other methods will just be stubs if they are designed for regular use. Second/third level inheritors will have to figure out their own thing. Should `B` call `A`'s method? That's up to `B`. And `A` can decide what if anything it overrides vs. extends from `plain`. The only difference is that `plain` doesn't have to worry about calling stuff from `base`.

Reply:
> (Also I don't see how it related to "language support")

It doesn't, all that stuff is in #2248 for merging to master and unrelated to the language module re-org. The commits are showing up here because I have this targeting develop for now, although that actually may change if I can work out shims.

Reply:
But then, here, in this very example, base -> plain ->  myclass.

Shouldn't our myclass invoke the parent options (= we inherit from plain, not base)

Reply:
(I mean, the example called `base.declareOptions` here, but this was probably overlooked for `plain.declareOptions` no? The `base`variable isn't even defined earlier, unless mistaken)

Reply:
Correct on both counts. I didn't catch it because it was incorrectly marked as running `base.<method>`, but the one it actually needed to run was the actual parent class `plain.<method>`. Fixed (here for the moment, but that fixup will end up in the other PR eventually, not this one).

Reply:
Fix cherry picked and applied as fixup in #2248, thanks for the catch.